pupil_src/shared_modules/video_overlay/workers/overlay_renderer.py

```diff
@@ -39,8 +39,21 @@ def draw_on_frame(self, target_frame):
             return
         overlay_frame = self.video.closest_frame_to_ts(target_frame.timestamp)
         overlay_image = overlay_frame.img
+        try:
+            is_fake_frame = overlay_frame.is_fake
+        except AttributeError:
```

Comment:
another way would be to check if `overlay_frame` is an instance of `Frame.FakeFrame`, since only an instance of this class should be considered a "fake frame". this would remove any possibility of getting an instance which does not have the `is_fake` attribute.

Reply:
@romanroibu We thought about this, too, but came to the conclusion that it the attribute implementation would be more robust if we changed the frame class hierarchy at some point.